airflow/__main__.py

```diff
@@ -24,12 +24,21 @@
 
 import argcomplete
 
+# The configuration module initializes and validates the conf object as a side effect the first
+# time it is imported. If it is not imported before importing the settings module, the conf
+# object will then be initted/validated as a side effect of it being imported in settings,
+# however this can cause issues since those modules are very tightly coupled and can
+# very easily cause import cycles in the conf init/validate code (since downstream code from
+# those functions likely import settings).
+# Therefore importing configuration early (as the first airflow import) avoids
+# any possible import cycles with settings downstream.
+from airflow import configuration
```

Comment:
Why not import `conf` directly here?

Reply:
It was making ruff upset doing it that way. Using `from airflow import ...` allows it to be the first import in the list without linters being angry.